srcCpp/perftest_publisher.cxx

```diff
@@ -1783,9 +1780,40 @@ int perftest_cpp::Publisher()
 
     message.size = INITIALIZE_SIZE;
 
-    for (unsigned long i = 0;
-            i < (std::max)(_InstanceCount, announcementSampleCount);
-            i++) {
+    /*
+     * Initial burst of data:
+     *
+     * The purpose of this initial burst of Data is to ensure that most
+     * memory allocations in the critical path are done before the test begings,
+     * for both the Writer and the Reader that receives the samples.
+     * It will also serve to make sure that all the instances are registered
+     * in advance in the subscriber application.
+     */
+
+    /*
+     * We initialize the number of samples sent to the maximum between instances
+     * to be sent and the minimum number of announcement messages we want to
+     * send.
+     */
+    unsigned long initializeSampleCount = (std::max)(
+            _InstanceCount,
+            announcementSampleCount);
+
+    /*
+     * We query the MessagingImplementation class to get the suggested sample
+     * count that we should send. This number might be based on the reliability
+     * protocol implemented by the middleware behind.
+     */
+    initializeSampleCount = (std::max)(
```

Comment:
I will not use announcementSampleCount anymore. Because we have the minimum (50) comming from the function GetInitializationSampleCount().

    unsigned long initializeSampleCount = (std::max)(
            _MessagingImpl->GetInitializationSampleCount(),
            initializeSampleCount);

Reply:
Done.